File: SourceCode/AspCoreVersion/src/ShopAware.Core/DataObjects/SearchResultDrugEventItem.cs
using System.Collections.Generic;

namespace ShopAware.Core
{
    namespace DataObjects
    {
        public class SearchResultDrugEventItem
        {
            #region Properties

            public List<string> BrandName { get; set; }

            public List<string> GenericName { get; set; }

            public List<string> ManufacturerName { get; set; }

            public List<string> Route { get; set; }

            public string BrandNamesString
            {
                get
                {
                    return string.Join(", ", BrandName.ToArray());
                }
                set
                {
                }
            }

            public string GenericNamesString
            {
                get
                {
                    return string.Join(", ", GenericName.ToArray());
                }
                set
                {
                }
            }

            #endregion

            #region Constructors

            public SearchResultDrugEventItem()
            {
                // VBConversions Note: Non-static class variable initialization is below.  Class variables cannot be initially assigned non-static values in C#.
                BrandName = new List<string>();
                GenericName = new List<string>();
                ManufacturerName = new List<string>();
                Route = new List<string>();
            }

            #endregion
        }
    }
}

File: SourceCode/Projects/open.fda.gov/ApiDalc/DataObjects/SearchResultDrugEventItem.vb
﻿#Region " Imports "

Imports ApiDalc.Enumerations
Imports Newtonsoft.Json.Linq ' OpenSource

#End Region

Namespace DataObjects

    Public Class SearchResultDrugEventItem

        Public Property BrandName As New List(Of String)
        Public Property GenericName As New List(Of String)
        Public Property ManufacturerName As New List(Of String)
        Public Property Route As New List(Of String)

        Public Property BrandNamesString As String
            Get
                Return Join(BrandName.ToArray, ", ")
            End Get
            Set(value As String)
            End Set
        End Property

        Public Property GenericNamesString As String
            Get
                Return Join(GenericName.ToArray, ", ")
            End Get
            Set(value As String)
            End Set
        End Property

    End Class

End Namespace
